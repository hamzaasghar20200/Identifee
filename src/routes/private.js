import { useState, useEffect } from 'react';
import { Route, Redirect } from 'react-router-dom';

import AuthService from '../services/auth.service';
import Layout from '../components/layout';
import userService from '../services/user.service';
import useCheckTokenValidity from '../hooks/useCheckTokenValidity';
import useSessionExpiryModal from '../components/modal/SessionExpiryModal';

const routerRedirect = {
  '/settings/resources/lesson': '/settings/resources',
  '/settings/resources/courses': '/settings/resources',
  '/settings/users/roles': '/settings/users',
  '/settings/users/groups': '/settings/users',
  '/contacts/:id/profile': '/contacts',
  '/contacts/organizations': '/contacts',
  '/contacts/:id/organization/profile': '/contacts',
  '/contacts/:organizationId/organization/profile/activity/:activityId':
    '/contacts',
};

const PrivateRoute = ({
  requireAdminAccess,
  isSplitView,
  component: Component,
  ...rest
}) => {
  const [adminAccess, setAdminAccess] = useState(undefined);
  const { SessionExpiryModal, setShowModal } = useSessionExpiryModal();

  useEffect(() => {
    userService
      .getUserInfo()
      .then((result) =>
        setAdminAccess(
          result?.role?.admin_access || result?.role?.owner_access || false
        )
      );
  }, []);

  useCheckTokenValidity(setShowModal);

  const privateRenderer = (props) => {
    const pathname = window.location.pathname;
    let privateRender = <Redirect to={`/login?redirect_uri=${pathname}`} />;

    if (AuthService.isTokenValid()) {
      if (!requireAdminAccess || (requireAdminAccess && adminAccess === true)) {
        const { location } = props;

        // eslint-disable-next-line no-prototype-builtins
        const hasRouteRedirect = routerRedirect.hasOwnProperty(
          location.pathname
        );

        if (hasRouteRedirect) {
          return <Redirect to={routerRedirect[location.pathname]} />;
        }

        privateRender = (
          <Layout isSplitView={isSplitView}>
            <SessionExpiryModal />
            <Component {...props} {...rest} />
          </Layout>
        );
      } else if (adminAccess === false) {
        privateRender = <Redirect to="/" />;
      } else {
        privateRender = null;
      }
    }

    return privateRender;
  };

  return <Route {...rest} render={privateRenderer} />;
};

export default PrivateRoute;
