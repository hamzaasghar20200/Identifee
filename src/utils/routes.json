{
  "accounts": "/accounts",
  "pipeline": "/pipeline",
  "pipeline2": "/pipeline",
  "prospects": "/prospects",
  "prospectsCompany": "/prospects/company/:companyId",
  "insights": "/reporting",
  "reporting": "/reporting",
  "tenant": "/tenants",
  "settings": "/settings",
  "Activities": "/activities",
  "training": "/settings/training",
  "learn": "/settings/learn",
  "users": "/settings/users",
  "workflow": "/settings/workflow",
  "usersProfile": "/settings/profile/users",
  "roles": "/settings/users/roles",
  "ManageRoles": "/settings/users/manage-roles",
  "contacts": "/contacts",
  "companies": "/companies",
  "contactProfile": "/contacts/:id/profile",
  "organizationProfile": "/contacts/:id/organization/profile",
  "dealsPipeline": "/pipeline",
  "errors": { "Unauthorized": "/403" },
  "groups": "/settings/users/groups",
  "lessons": "/learn/my-lessons",
  "favorites": "/learn/my-favorites",
  "selfAssessment": "/learn/self-assessment",
  "selfAssessmentPublic": "/self-assessment",
  "selfAssessmentAdaption": "/adaption-self-assessment",
  "lesson": "/learn/lessons/:id",
  "courses": "/learn/courses",
  "learnLessons": "/learn/lessons",
  "categories": "/learn/categories",
  "importOrg": "/settings/bulk-import/companies",
  "importPeople": "/settings/bulk-import/people",
  "importDeals": "/settings/bulk-import/deals",
  "integrations": "/settings/integrations",
  "pipelinesAndStages": "/settings/pipelines-and-stages",
  "fields": "/settings/fields",
  "branding": "/settings/branding",
  "resources": "/prospecting",
  "prospecting": "/prospecting",
  "resourcesOrganization": "/resources/:name/organization/details",
  "notificationsAll": "/notifications/all",
  "maintenance": "/maintenance",
  "welcome": "/welcome",
  "insightsCompanies": "/insights",
  "learnMain": "/learn",
  "aiAssist": "/ai-assist",
  "videoPlayer": "/video"
}
