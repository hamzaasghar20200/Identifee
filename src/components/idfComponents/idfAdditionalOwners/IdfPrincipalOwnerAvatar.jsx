import { OverlayTrigger, Tooltip } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import { PRINCIPAL_OWNER_LABEL } from '../../../utils/constants';

import Avatar from '../../Avatar';
import './IdfAdditionalOwners.css';

const IdfPrincipalOwnerAvatar = ({
  sizeIcon,
  mainOwner,
  defaultSize,
  className,
  isClickable = true,
}) => {
  return (
    <div className={`d-flex main-owner ${className}`}>
      <OverlayTrigger
        key={mainOwner?.id}
        placement="bottom"
        overlay={
          <Tooltip
            id={`tooltip-${mainOwner?.id}`}
            className={`tooltip-profile font-weight-bold`}
          >
            <p>{`${mainOwner?.first_name} ${mainOwner?.last_name}`}</p>
            <p>{PRINCIPAL_OWNER_LABEL}</p>
          </Tooltip>
        }
      >
        <div className="border-0">
          {' '}
          {isClickable ? (
            <span className="d-flex">
              <Avatar
                classModifiers="avatar-md"
                user={mainOwner}
                defaultSize={defaultSize}
                sizeIcon={sizeIcon}
              />
            </span>
          ) : (
            <Link className="d-flex cursor-default">
              <Avatar
                classModifiers="avatar-md"
                user={mainOwner}
                defaultSize={defaultSize}
                sizeIcon={sizeIcon}
              />
            </Link>
          )}
        </div>
      </OverlayTrigger>
    </div>
  );
};

export default IdfPrincipalOwnerAvatar;
