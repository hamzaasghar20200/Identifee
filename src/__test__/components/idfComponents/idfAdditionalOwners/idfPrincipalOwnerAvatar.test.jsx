import { fireEvent, render } from '@testing-library/react';
import { waitFor } from '@testing-library/dom';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import IdfPrincipalOwnerAvatar from '../../../../components/idfComponents/idfAdditionalOwners/IdfPrincipalOwnerAvatar';
import { usersProfile } from '../../../../utils/routes.json';

describe('Test IdfPrincipalOwnerAvatar', () => {
  const props = {
    withoutRequest: true,
    mainOwner: {
      id: 1,
      first_name: 'First Name',
      last_name: 'Last Name',
      title: 'Title',
    },
  };

  const history = createMemoryHistory();

  beforeEach(async () => {
    jest.clearAllMocks();
    jest.resetModules();
    jest.restoreAllMocks();
    jest.resetAllMocks();
  });

  it('Snapshot and render showing the initials of the name', () => {
    const component = render(
      <Router history={history}>
        <IdfPrincipalOwnerAvatar {...props} />
      </Router>
    );
    expect(component.container).toBeVisible();
    expect(component.container).toHaveTextContent('FL');
    expect(component.baseElement).toMatchSnapshot();
  });

  it("Shouldn't allow other initials differents to the name", () => {
    const { container } = render(
      <Router history={history}>
        <IdfPrincipalOwnerAvatar {...props} />
      </Router>
    );
    expect(container).toBeVisible();
    expect(container).not.toHaveTextContent('NA');
  });

  it('Should render the avatar with the default size', () => {
    const size = 'sm';
    const { container } = render(
      <Router history={history}>
        <IdfPrincipalOwnerAvatar {...props} />
      </Router>
    );

    const avatar = container.getElementsByClassName(`avatar-${size}`)[0];

    expect(avatar).toBeVisible();
    expect(avatar).toHaveClass(`avatar-${size}`);
  });

  it('Should render the avatar with the size xs', () => {
    const size = 'xs';
    const { container } = render(
      <Router history={history}>
        <IdfPrincipalOwnerAvatar {...props} defaultSize={size} />
      </Router>
    );

    const avatar = container.getElementsByClassName(`avatar-${size}`)[0];

    expect(avatar).toBeVisible();
    expect(avatar).toHaveClass(`avatar-${size}`);
    expect(avatar).not.toHaveClass(`avatar-sm`);
  });

  it('Should render the tooltip when the mouse is over component', async () => {
    const { mainOwner } = props;
    const firstText = `${mainOwner.first_name} ${mainOwner.last_name} - ${mainOwner.title}`;
    const component = render(
      <Router history={history}>
        <IdfPrincipalOwnerAvatar {...props} />
      </Router>
    );

    const span = component.getByText('FL');
    fireEvent.mouseOver(span);

    const tool = component.getByText(firstText);

    await waitFor(() => {
      expect(tool).toBeVisible();
      expect(tool).toHaveTextContent(firstText);
    });
  });

  it('Should call the link when click the component', async () => {
    const { mainOwner } = props;
    history.push = jest.fn();

    const component = render(
      <Router history={history}>
        <IdfPrincipalOwnerAvatar {...props} />
      </Router>
    );

    const link = component.container.querySelector(`.main-owner > a`);
    fireEvent.click(link);

    expect(history.push).toHaveBeenCalledTimes(1);
    expect(history.push).toHaveBeenCalledWith(
      `${usersProfile}/${mainOwner.id}`
    );
  });
});
