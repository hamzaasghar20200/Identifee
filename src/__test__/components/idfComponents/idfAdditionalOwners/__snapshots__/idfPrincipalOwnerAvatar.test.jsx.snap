// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test IdfPrincipalOwnerAvatar Snapshot and render showing the initials of the name 1`] = `
<body>
  <div>
    <div
      class="d-flex main-owner undefined"
    >
      <a
        class="d-flex"
        href="/settings/profile/users/1"
      >
        <div
          class="avatar avatar-sm avatar-circle avatar-soft-primary avatar-md"
        >
          <span
            class="avatar-initials"
          >
            FL
          </span>
        </div>
      </a>
    </div>
  </div>
</body>
`;
